app/assets/javascripts/launchpad_keep_alive.coffee

```diff
@@ -0,0 +1,53 @@
+$(document).ready ->
+  # Only do the keep alive if launchpad is enabled and the user isn't on
+  # the welcome page
+  if launchpadEnabled && !$('main.welcome').length > 0
+    fullSessionLength = 840000 # 14 minutes (in milliseconds)
+    lastActiveTime = Date.now()
+
+    # sessionStart is the beginning of the session in which the user was last active
+    sessionStart = Date.now()
+
+    # if the user changes pages in the middle of the session window
+    # the keep alive timer will need to be shortened to ensure
+    # the keep alive call is successful
+    sessionLength = if remainingSessionTime > sessionLength then sessionLength else remainingSessionTime
```

Comment:
since this isn't declared (since I thought we could combine the statements) I had thought this might need to be changed to `sessionLength = if remainingSessionTime > fullSessionLength then fullSessionLength else remainingSessionTime`

Reply:
https://gph.is/1W2P1V1